shoes-swt/lib/shoes/swt/shoes_layout.rb

```diff
@@ -4,42 +4,54 @@ class ShoesLayout < ::Swt::Widgets::Layout
       attr_accessor :gui_app
 
       def layout(*_dontcare)
-        dsl_app = @gui_app.dsl
-        height = dsl_app.height
-        scrollable_height = dsl_app.top_slot.contents_alignment
-        set_gui_size(height, scrollable_height)
-
-        dsl_app.top_slot.width  = dsl_app.width
-        dsl_app.top_slot.height = [scrollable_height, height].max
+        height, scroll_height, is_scrolling = gather_height_info(@gui_app.dsl)
+        set_heights(@gui_app.dsl, @gui_app.real, height, scroll_height)
 
-        vertical_bar = @gui_app.shell.getVerticalBar
-        vertical_bar.setVisible(scrollable_height > height)
-        if scrollable_height > height
-          handle_scroll_bar(vertical_bar, height, scrollable_height)
+        if is_scrolling
+          show_scrollbar(vertical_scrollbar, height, scroll_height)
         else
-          set_gui_location
+          hide_scrollbar(vertical_scrollbar, @gui_app.real)
         end
       end
 
-      private
```

Comment:
That private is a smell that there's probably another object or so hiding in the class was one of the first and major things Steve taught me about programming back in 2011 - good times :)

Tend to use `private` quite a bit myself as I heavily extract even one line methods to highlight a concept or isolate change to that one method :) Still, all approaches valid :)